jdbc/src/main/clojure/xtdb/next/jdbc.clj

```diff
@@ -20,11 +20,22 @@
                    (String. StandardCharsets/UTF_8)))))
 
 (defn- denormalize-keys [v, ^IKeyFn key-fn]
-  (w/prewalk (fn [v]
-               (cond-> v
-                 (instance? Map v)
-                 (update-keys #(.denormalize key-fn %))))
-             v))
+  (letfn [(process-map [m]
+            (if (keyword? (first (keys m)))
+              m
+              (update-keys m #(.denormalize key-fn %))))]
+    (w/prewalk (fn [v]
+                 (cond-> v
+                   (instance? Map v)
+                   process-map
+
+                   (instance? Collection v)
+                   (->>
```

Comment:
in theory, given it's a pre-walk, we should be able to just have `(instance? List v) vec` here, and have the pre-walk deal with iterating it?

(also worth us adding `(instance? Set v) set`, if the cause of this one was prewalk not recognising Java collections)